src/model/block/payload/transaction/output/mod.rs

```diff
@@ -261,6 +263,46 @@ impl From<Output> for iota::dto::OutputDto {
     }
 }
 
+/// A [`Tag`] associated with an [`Output`].
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct Tag(#[serde(with = "bytify")] Vec<u8>);
+
+impl Tag {
+    /// Creates a [`Tag`] from `0x`-prefixed hex representation.
+    pub fn from_hex<T: AsRef<str>>(tag: T) -> Result<Self, prefix_hex::Error> {
+        Ok(Self(prefix_hex::decode::<Vec<u8>>(tag.as_ref())?))
+    }
+
+    /// Converts the [`Tag`] to its `0x`-prefixed hex representation.
+    pub fn to_hex(&self) -> String {
+        prefix_hex::encode(&*self.0)
+    }
+}
+
+// Note: assumes an ASCII string as input.
```

Comment:
This should say utf-8, not ascii, right?

Reply:
I am not sure ... `String`s obviously support `utf-8`, but I think IOTA assumes tags to be ASCII.

Reply:
Well the fact is rust strings are not ASCII, so this comment is odd. Since we only use it for testing, I think we should just remove the comment.